lib/Doctrine/Persistence/Mapping/AbstractClassMetadataFactory.php

```diff
@@ -45,23 +59,54 @@ abstract class AbstractClassMetadataFactory implements ClassMetadataFactory
     /**
      * Sets the cache driver used by the factory to cache ClassMetadata instances.
      *
+     * @deprecated Deprecated in favor of setCache
+     *
      * @return void
      */
     public function setCacheDriver(?Cache $cacheDriver = null)
     {
+        @trigger_error(sprintf('%s is deprecated. Use setCache() with a PSR-16 cache instead.', __METHOD__), E_USER_DEPRECATED);
+
         $this->cacheDriver = $cacheDriver;
+
+        if ($cacheDriver === null) {
+            $this->cache = null;
+
+            return;
+        }
+
+        if (! $cacheDriver instanceof CacheProvider) {
+            throw new BadMethodCallException('Cannot convert cache to PSR-6 cache');
```

Comment:
:thinking: this looks like a BC-break

Reply:
Yeah, it's the one edge case we can't really do anything about. I'm willing to wait for someone to complain...

Reply:
Shall we make a note about this in UPGRADE.md?

Reply:
Good point. Unresolving this conversation and will follow up.